src/datatable/xlsx.py

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env python3
+#-------------------------------------------------------------------------------
+# Copyright 2018-2021 H2O.ai
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#-------------------------------------------------------------------------------
+import datatable as dt
+
+
+
+def read_xlsx_workbook(filename, subpath):
+    try:
+        import openpyxl
+    except ImportError:
+        raise dt.exceptions.ImportError(
+            "Module `openpyxl` is required in order to read Excel file '%s'"
+            % filename)
+
+    if subpath:
+        wb = openpyxl.load_workbook(filename, data_only=True)
+        range2d = None
+        if subpath in wb.sheetnames:
+            sheetname = subpath
+        else:
+            if "/" in subpath:
```

Comment:
Appveyor is failing in Windows because of the conversion to a path with backslashes.

1. `C:\Path\to\file.xlsx\Sheet\A2:B9` fails (also for xls files with the current xlrd-based parser).
2. `C:\Path\to\file.xlsx\Sheet/A2:B9` also fails, since there is not such a `C:\Path\to\file.xlsx\Sheet` sheet, aftersplitting.
3. `C:\Path\to\file.xlsx/Sheet/A2:B9` works.

The intended behavior in Windows is to work with (1), right? I guess both xlsx.py and xls.py should check for backslashes if no `(sheetname, range2d)` pair was found when a subpath is specified.

Reply:
Windows support was added after the excel reading feature, so, you're right, we probably missed the backslash issue. My feeling is that on Windows we should support both types of slashes. What do you think?

Reply:
Yes, I guess it would be more consistent for the user if they do not have to change the code for different platforms. I'll give it a try in a day.

Reply:
This issue has been fixed as of https://github.com/h2oai/datatable/pull/3220